src/core/thread/link_metrics_tlvs.hpp

```diff
@@ -209,15 +209,86 @@ class QueryOptionsSubTlv : public Tlv, public TlvInfo<SubTlv::kQueryOptions>
 
 } OT_TOOL_PACKED_END;
 
+/**
+ * This class defines Link Metrics Forward Probing Registration Sub-TLV.
+ *
+ */
 OT_TOOL_PACKED_BEGIN
-class EnhAckConfigSubTlv : public Tlv, public TlvInfo<SubTlv::kEnhAckConfig>
+class FwdProbingRegSubTlv : public Tlv, public TlvInfo<SubTlv::kFwdProbingReg>
 {
 public:
+    static constexpr uint8_t kMinLength = sizeof(uint8_t) + sizeof(SeriesFlags); ///< Minimum expected TLV length
+
     /**
-     * Default constructor
+     * This method initializes the TLV.
      *
      */
-    EnhAckConfigSubTlv(void) { Init(); }
+    void Init(void)
+    {
+        SetType(SubTlv::kFwdProbingReg);
+        SetLength(kMinLength);
+    }
+
+    /**
+     * This method indicates whether or not the TLV appears to be well-formed.
+     *
+     * @retval true   The TLV appears to be well-formed.
+     * @retval false  The TLV does not appear to be well-formed.
+     *
+     */
+    bool IsValid(void) const { return GetLength() >= kMinLength; }
+
+    /**
+     * This method gets the Forward Series ID value.
+     *
+     * @returns The Forward Series ID.
+     *
+     */
+    uint8_t GetSeriesId(void) const { return mSeriesId; }
+
+    /**
+     * This method sets the Forward Series ID value.
+     *
+     * @param[in] aSeriesId  The Forward Series ID.
+     *
+     */
+    void SetSeriesId(uint8_t aSeriesId) { mSeriesId = aSeriesId; }
+
+    /**
+     * This method gets the Forward Series Flags.
+     *
+     * @returns The Forward Series Flags.
+     *
+     */
+    SeriesFlags &GetSeriesFlags(void) { return mSeriesFlags; }
```

Comment:
Given that `SeriesFlags` is a `uint8_t`, is there any real benefit to returning a reference? I'm fine with keeping this as-is since it's not obvious that it is just a `uint8_t`.

Reply:
The reason for reference here so we can use `SeriesFlag` method `SetFrom()` on it when we prepare the TLV.

That said, I fully agree with you point on this. I also like it better if we treat these as simple `uint8_t` as well (same situation with `TypeIdFlags` which is again a wrapper over a `uint8_t`). I hope/intend to change these later in follow-up PRs (when I get the chance). We can still define a bunch of helper methods/functions to  get/set different bits/flags in these or calculate them from other representations.